stdlib/public/core/StringStorage.swift

```diff
@@ -237,14 +297,38 @@ final internal class _StringStorage: _AbstractStringStorage {
 
     return nil
   }
+  
+  @objc(UTF8String)
+  @_effects(readonly)
+  final internal func _utf8String() -> UnsafePointer<UInt8>? {
+    if let contents = asGuts.fastUTF8 {
+      return contents
+    }
+    return _cocoaUTF8StringTrampoline(self)
+  }
+
+  @objc(cStringUsingEncoding:)
+  @_effects(readonly)
+  final internal func _cString(encoding: UInt) -> UnsafePointer<UInt8>? {
+    return _cStringUsingEncoding(self, encoding)
+  }
+  
+  @objc(getCString:maxLength:encoding:)
+  @_effects(releasenone)
+  final internal func getCString(_ bufferPtr: UnsafeMutablePointer<UInt8>?,
+                               maxLength: Int,
+                               encoding: UInt)
+    -> Int8 {
+     return _getCString(self, bufferPtr, maxLength, encoding)
+  }
```

Comment:
Any reason for the out-of-line calls over having the implementation here directly?

Reply:
Just deduplicating code